app/lib/package/backend.dart

```diff
@@ -1203,11 +1203,24 @@ class PackageBackend {
           'publishing from github is not enabled');
     }
 
-    // Repository must be set and matching the action's repository.
+    // verify that fields are configured
     final repository = githubPublishing!.repository;
-    if (repository == null ||
-        repository.isEmpty ||
-        repository != agent.payload.repository) {
+    if (repository == null || repository.isEmpty) {
+      throw ArgumentError('Missing or empty repository.');
+    }
+    final tagPattern = githubPublishing.tagPattern ?? '';
+    if (!tagPattern.contains('{{version}}')) {
+      throw ArgumentError(
+          'Configured tag pattern does not include `{{version}}`');
+    }
+    final requireEnvironment = githubPublishing.requireEnvironment ?? false;
+    final environment = githubPublishing.environment;
+    if (requireEnvironment && (environment == null || environment.isEmpty)) {
+      throw ArgumentError('Missing or empty environment.');
+    }
```

Comment:
ArgumentError becomes a 5xx, right?

I guess that's fine. We agree this should never happen, right?

We could also throw `AssertionError` instead.

Reply:
Yeah, we should use AssertionError instead.